src/bun.js/bindings/webcore/JSWasmStreamingCompiler.cpp

```diff
@@ -0,0 +1,238 @@
+}
+
+const ClassInfo JSWasmStreamingCompiler::s_info = { "WasmStreamingCompiler"_s, &Base::s_info, nullptr, nullptr, CREATE_METHOD_TABLE(JSWasmStreamingCompiler) };
+
+JSWasmStreamingCompiler::JSWasmStreamingCompiler(Structure* structure, JSDOMGlobalObject& globalObject, Ref<Wasm::StreamingCompiler>&& impl)
+    : JSDOMWrapper<Wasm::StreamingCompiler>(structure, globalObject, WTFMove(impl))
+{
+}
+
+void JSWasmStreamingCompiler::finishCreation(VM& vm)
+{
+    Base::finishCreation(vm);
+    ASSERT(inherits(info()));
+}
+
+JSObject* JSWasmStreamingCompiler::createPrototype(VM& vm, JSDOMGlobalObject& globalObject)
+{
+    auto* structure = JSWasmStreamingCompilerPrototype::createStructure(vm, &globalObject, globalObject.objectPrototype());
+    structure->setMayBePrototype(true);
+    return JSWasmStreamingCompilerPrototype::create(vm, &globalObject, structure);
+}
+
+JSObject* JSWasmStreamingCompiler::prototype(VM& vm, JSDOMGlobalObject& globalObject)
+{
+    return getDOMPrototype<JSWasmStreamingCompiler>(vm, globalObject);
+}
+
+void JSWasmStreamingCompiler::destroy(JSCell* cell)
+{
+    auto* thisObject = static_cast<JSWasmStreamingCompiler*>(cell);
+    thisObject->JSWasmStreamingCompiler::~JSWasmStreamingCompiler();
+}
+
+static inline EncodedJSValue jsWasmStreamingCompilerPrototypeFunction_addBytesBody(JSGlobalObject* lexicalGlobalObject, CallFrame* callFrame, typename IDLOperation<JSWasmStreamingCompiler>::ClassParameter castedThis)
+{
+    auto& vm = JSC::getVM(lexicalGlobalObject);
+    auto throwScope = DECLARE_THROW_SCOPE(vm);
+    auto& impl = castedThis->wrapped();
+
+    auto chunkValue = callFrame->uncheckedArgument(0);
+
+    // See getWasmBufferFromValue in JSC's JSWebAssemblyHelpers.h
+    if (auto arrayBufferView = jsDynamicCast<JSArrayBufferView*>(chunkValue)) {
+        if (isTypedArrayType(arrayBufferView->type())) {
+            validateTypedArray(lexicalGlobalObject, arrayBufferView);
+            RETURN_IF_EXCEPTION(throwScope, {});
+        } else {
+            // DataView
+            IdempotentArrayBufferByteLengthGetter<std::memory_order_relaxed> getter;
+            if (!jsCast<JSDataView*>(arrayBufferView)->viewByteLength(getter)) [[unlikely]] {
+                throwTypeError(lexicalGlobalObject, throwScope, typedArrayBufferHasBeenDetachedErrorMessage);
+                return {};
+            }
+        }
+
+        impl.addBytes({ static_cast<const uint8_t*>(arrayBufferView->vector()), arrayBufferView->byteLength() });
+        return encodedJSUndefined();
+    } else if (auto arrayBuffer = jsDynamicCast<JSArrayBuffer*>(chunkValue)) {
+        auto arrayBufferImpl = arrayBuffer->impl();
+        if (arrayBufferImpl->isDetached()) {
+            throwTypeError(lexicalGlobalObject, throwScope, typedArrayBufferHasBeenDetachedErrorMessage);
+            return {};
+        }
+
+        impl.addBytes({ static_cast<const uint8_t*>(arrayBufferImpl->data()), arrayBufferImpl->byteLength() });
```

Comment:
```suggestion
        impl.addBytes(arrayBufferImpl->span());
```

Reply:
Huh, could've sworn that didn't work before...